packages/apollo/src/commands/service/check.ts

```diff
@@ -602,32 +597,77 @@ export default class ServiceCheck extends ProjectCommand {
         )
       );
     } else if (shouldOutputMarkdown) {
-      const serviceName = config.service && config.service.name;
+      // This _should_ always be here; but TypeScript tells us that's optional. If we check it here, then
+      // passing `config` to any other function will signify that `config.service` might now be null or
+      // undefined. Save it as a const to tell TypeScript `service` can't be changed.
+
+      const { service } = config;
+      if (!service) {
+        throw new Error(
+          "Service mising from config. This should have been validated elsewhere"
+        );
+      }
 
-      if (!serviceName) {
+      const graphName = config.service && config.service.name;
```

Comment:
This might be a nice place for a comment explaining the naming conventions as a result of federation / the mismatch between this and the gateway!